torchgeo/datasets/geo.py

```diff
@@ -289,6 +289,8 @@ def files(self) -> set[str]:
         # Make iterable
         if isinstance(self.paths, str):
             paths: Iterable[str] = [self.paths]
+        elif isinstance(self.paths, os.PathLike):
+            paths: Iterable[os.PathLike] = [self.paths]
```

Comment:
Can probably keep this to a single if-statement. Either check both or check "not iterable"

Reply:
Yes, you are correct sir, will implement this.